### ISSUE TITLE
🐛 [Bug Report] Character can learn two different fishing skills.

### ISSUE THREAD

**Issue by aleigood** [cmangos/issues#2438]
## 🐛 Bugreport
You can learn two different fishing skills from NPC trainer, and fishing can be done without a fishing rod.

![snipaste_20210330_224736](https://user-images.githubusercontent.com/3367999/113009558-a3974600-91aa-11eb-9fb1-feae51df7f24.jpg)

### Version & Environment
Client Version: 
"3.3.5a" (WOTLK)

CMaNGOS Repo & Commit Hash: 
eeeb560413b8282c9e996deb1c6b0c108b2224e3

Database Repo & Commit Hash: 
fa30e3d2eba4671eb413e966defeed72d1a25381

Operating System: 
Linux 
### Steps to reproduce
1. new character
2. .character level 10
3. .go cr 15321
4. learn Apprentice Fishing
5. .setskill 356 70
6. learn Journeyman Fishing

**Comment by al3xc1985** [comment]
But as far as I tested until now, if u fish by normal play, this bug does not exist.

Have u tried that way?

**Comment by aleigood** [comment]
> But as far as I tested until now, if u fish by normal play, this bug does not exist.
> 
> Have u tried that way?

Yep, I upgrade the Apprentice Fishing by normal play, the above steps are just for the convenience of others to reproduce the bug.

In addition it's require the designated trainer 15321.

**Comment by al3xc1985** [comment]
same happends in some circumstances with raiding skill

**Comment by al3xc1985** [comment]
The bug does not exist anymore @AnonXS I tested several times

**Comment by al3xc1985** [comment]
I followed in my test exact same steps given by the aleigood
